### PR TITLE
Refactor bank get vote accounts 

### PR DESCRIPTION
**Pull request by sagar-solana**
#### Problem

Requiring a closure to filter out elements makes `bank::vote_accounts` and `accounts::get_vote_accounts` unnecessarily complicated. Instead returning an iterator allows for much more flexibility.

#### Summary of Changes

Refactored the two fns mentioned above, and cleaned up `leader_confirmation_service::get_last_supermajority_timestamp`

### CODE REVIEW

**Comment by sagar-solana** [comment]
@carllin, thanks for simplifying this.

File: src/leader_confirmation_service.rs [hunk 1]
```diff
@@ -28,26 +28,23 @@ pub struct LeaderConfirmationService {
 impl LeaderConfirmationService {
     fn get_last_supermajority_timestamp(
         bank: &Arc<Bank>,
-        leader_id: Pubkey,
```
Comment (rob-solana) [inline src/leader_confirmation_service.rs]: is there no test for this "exclude the leader from our calculation" case that's broken by this change?

**Comment by rob-solana** [comment]
this is better ;)

File: src/leader_confirmation_service.rs (hunk 1 above)
Comment (sagar-solana) [reply src/leader_confirmation_service.rs]: Oh there is a test apparently. Something failed. I'll get to it later tonight. Afk now :)

**Pull request merged**